cpp/src/CMakeLists.txt

```diff
@@ -521,6 +243,10 @@ file(MAKE_DIRECTORY ${root_directory}/releases)
 add_library(spark_columnar_jni SHARED ${SPARK_COLUMNAR_PLUGIN_SRCS})
 add_dependencies(spark_columnar_jni jni_proto)
 
+find_arrow_lib(${ARROW_LIB_NAME})
```

Comment:
It looks like two times to call find_arrow_lib function in "cpp/src/CMakeLists.txt" and "cpp/gazelle-cpp/src/CMakeLists.txt"
I can see below information has been printed two times in the unit test. 
-- Found Arrow Library: /home/runner/work/gluten/gluten/tools/build/arrow_install/lib/libarrow.so.800
-- Found Arrow Library: /home/runner/work/gluten/gluten/tools/build/arrow_install/lib/libarrow_dataset_jni.so.800
-- Found Arrow Library: /home/runner/work/gluten/gluten/tools/build/arrow_install/lib/libgandiva.so.800
In original code, we keep find_arrow_lib in cpp/src/CMakeLists.txt since multiple backend may need to use Arrow.
Check if we keep it in cpp/src/CMakeLists.txt and remove the function call in gazelle-cpp?

Reply:
Per discussed, the code separate arrow library check to two different parts: 
three libraries for general purpose(used for all other backends).
three libraries for gazelle-cpp only.
I'm ok for the change.